lib/kube_deploy_tools/artifact_registry/driver_gcs.rb

```diff
@@ -34,6 +34,9 @@ def get_registry_artifact_path(name:, flavor:, project:, build_number:)
       # change where all past clients will not be able to download new builds and
       # new clients will not be able to download old builds. Change with caution.
       #
+      if flavor.empty? && name.empty? && build_number.empty?
+        return "#{@bucket}/project/#{project}/build/"
```

Comment:
hmm - this is a little sus for me - I guess the way I would do it is create a common method used by get_registry_artifact_path and get_latest_build_number that returns the build directory - then flag both that new method and get_registry_artifact_path as "here be dragons, don't change or shit breaks"

Instead of creating divergent code paths, you're instead creating composition of code paths.

Another way to see it is that there are very few scenarios where you might call get_registry_artifact_path without flavor, name and build_number set. So this is really just a special case that applies to your particular implementation of get_latest_build_number.

Reply:
That makes sense - thanks for the elaborating. I made the changes for the suggestion.